src/filepattern/cpp/util/util.hpp

```diff
@@ -481,6 +481,18 @@ namespace v {
         }
         return slice;
     }
+
+    inline bool areEqual(std::vector<std::string> vec1, std::vector<std::string> vec2){
```

Comment:
We are modifying the vectors (the order) while checking for equality. I suggest to pass a const ref vector and create a sorted on from them in the function. Also, we should revisit this to see if there is another better (computationally efficient) way to compare equality.